File: services/reservation/src/main/java/com/example/iter/reservation/domain/repository/ReturnReceiptRepository.java
package com.example.iter.reservation.domain.repository;

import com.example.iter.reservation.domain.entity.ReturnReceipt;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import java.util.List;
import java.util.Optional;

public interface ReturnReceiptRepository extends JpaRepository<ReturnReceipt, Long> {
    Optional<ReturnReceipt> findByRentalId(Long rentalId);

    // 여러 거래의 반납 증빙을 한 번에 조회합니다.
    List<ReturnReceipt> findAllByRental_IdIn(Collection<Long> rentalIds);
}

File: services/reservation/src/main/kotlin/iter/reservation/domain/repository/ReturnReceiptRepository.kt
package iter.reservation.domain.repository

import iter.reservation.domain.entity.ReturnReceipt
import org.springframework.data.jpa.repository.JpaRepository

import java.util.Optional

interface ReturnReceiptRepository : JpaRepository<ReturnReceipt, Long> {
    fun findByRentalId(rentalId: Long): Optional<ReturnReceipt>

    // 여러 거래의 반납 증빙을 한 번에 조회합니다.
    fun findAllByRental_IdIn(rentalIds: Collection<Long>): List<ReturnReceipt>
}
